File: palworld_info_server/src/structures/entities/structure.entity.ts
import { Column, Entity, PrimaryGeneratedColumn } from "typeorm";

export enum StructureType {
  "Product" = "Product",
  "Pal" = "Pal",
  "Food" = "Food",
  "Infrastructure" = "Infrastructure",
  "Light" = "Light",
  "Foundation" = "Foundation",
  "Defense" = "Defense",
  "Other" = "Other",
  "Furniture" = "Furniture",
  "Dismantle" = "Dismantle",

}

@Entity()
export class StructureEntity {
  @PrimaryGeneratedColumn()
  id: number;

  @Column({ unique: true })
  name: string;

  @Column({ nullable: true })
  slug: string;

  @Column({ nullable: true })
  iconUrl: string;

  @Column({ nullable: true })
  summary: string;

  @Column({
    type: "enum",
    enum: StructureType
  })
  structureType: StructureType;

  @Column({ nullable: true })
  buildWork: number;

  @Column({ nullable: true })
  energyType: string;

  @Column({ nullable: true })
  consumeEnergySpeed: number;


}

File: palworld_info_app/lib/domains/structure_entity.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:palworld_info_app/utils/constants.dart';

part 'structure_entity.freezed.dart';
part 'structure_entity.g.dart';

@freezed
class StructureEntity with _$StructureEntity {

  const factory StructureEntity({
    int? id,
    String? name,
    String? slug,
    String? iconUrl,
    String? summary,
    String? structureType,
    int? buildWork,
    String? energyType,
    int? consumeEnergySpeed,
  }) = _StructureEntity;

  factory StructureEntity.fromJson(Map<String, dynamic> json) =>
      _$StructureEntityFromJson(json);
}